nservicebus/upgrades/7to8/index.md

```diff
@@ -26,6 +26,76 @@ The following provider packages will no longer be provided:
 * [Log4net](/nservicebus/logging/log4net.md)
 * [NLog](/nservicebus/logging/nlog.md)
 
+## CancellationToken support
+
+NServiceBus version 8 supports cooperative cancellation using `CancellationToken`.
+
+### Non-message-handling contexts
+
+Methods on extension points are updated to include a mandatory `CancellationToken` parameter. This includes abstract classes and interfaces needed to implement a message transport or persistence libary, as well as other extension points like `IDataBus`, `FeatureStartupTask`, `INeedToInstallSomething`. Implementors can be updated by adding the `CancellationToken` parameter to the end of the method signature.
+
+Methods used outside the message processing pipeline now include an optional `CancellationToken` parameter, including methods for starting and stopping endpoints, and methods to Send/Publish messages from outside a message processing pipeline, such as from within a web application.
+
+It is recommended to forward a `CancellationToken` within a method to any method that accepts one. For example, within a web application controller:
+
+```csharp
```

Comment:
Should this be a snippet?

Reply:
Yeah all these snippets can be snippets, but I figured it was easier to just write it out while WIP and snippetize later.